dashboard/test/models/peer_review_test.rb

```diff
@@ -75,8 +75,17 @@ class PeerReviewTest < ActiveSupport::TestCase
     # Assign one review
     PeerReview.last.update! reviewer: create(:user)
 
+    # Unsubmit the level
+    user_level = UserLevel.find_by(user: @user, level: @level, script: @script)
+    user_level.update(submitted: false)
+
+    # The unassigned review was destroyed
+    assert_equal 1, PeerReview.count - original_count
+
+    # Submit a new answer
     updated_level_source = create :level_source, data: 'UPDATED: My submitted answer'
 
+    # Two new reviews were created
     track_progress updated_level_source.id
     assert_equal 3, PeerReview.count - original_count
   end
```

Comment:
This test calls `track_progress` twice.  It started failing after this change because the `user_level.submitted` value didn't change on the second call, so the hook would not run and reviews would not be created.

I've updated the test to include an unsubmit step in the middle, which I believe is more representative of a real workflow for peer reviews.

I might be wrong about this though: Do we instead need to run the `after_submit` hook any time `submitted` is true and the level source changed? Or on some other conditions? There's a chance I'm simply wrong about this refactor and we have to isolate this logic to record changes that originate in the `track_level_progress_sync` method.  Interested in feedback.

Reply:
Update: @clareconstantine and I took a look at this together and it looks like changing a submitted reflection _is_ a valid use case that we need to handle. I've therefore undone the changes to this test (except for a few comments) after introducing a more precise `submitted_or_resubmitted?` condition under which the review-creating logic may run, so this test is back to passing in its original form.